.travis.yml

```diff
@@ -1,26 +1,57 @@
 sudo: required
+dist: trusty
 language: java
 
 services:
   - docker
 
-install:
-  # install heroku CLI
-  - wget -qO- https://toolbelt.heroku.com/install.sh | sh
-  # login to heroku
-  - echo "$HEROKU_PASSWORD" | docker login -u "$HEROKU_USERNAME" --password-stdin registry.heroku.com
-
 script:
   # build docker image
   - mvn clean install
   - docker build -t truthtree:test .
   - docker tag truthtree:test registry.heroku.com/truthtree/web
+  - docker tag truthtree:test 869798252720.dkr.ecr.us-west-2.amazonaws.com/truthtree:latest
+
+before_deploy:
+  # install heroku CLI
+  - wget -qO- https://toolbelt.heroku.com/install.sh | sh
+  # login to heroku
+  - echo "$HEROKU_PASSWORD" | docker login -u "$HEROKU_USERNAME" --password-stdin registry.heroku.com
+
+  # install ecs-deploy
+  - sudo apt-get update
+  - sudo apt-get install jq -y
+  - curl https://raw.githubusercontent.com/silinternational/ecs-deploy/master/ecs-deploy | sudo tee -a /usr/bin/ecs-deploy
+  - sudo chmod +x /usr/bin/ecs-deploy
+
+  # install python using conda
+  - >-
+    wget https://repo.anaconda.com/archive/Anaconda3-2018.12-Linux-x86_64.sh -O ~/anaconda3.sh
+    || true
+    && bash ~/anaconda3.sh -b -p ~/anaconda3 || true
+    && rm ~/anaconda3.sh
+    && echo 'export PATH="$HOME/anaconda3/bin:$PATH"' >> ~/.bashrc
+    && source ~/.bashrc
+    && conda create -n py27 python=2.7.16 -y
+    && source activate py27
+
+  # install AWS CLI
+  - pip install --user awscli
+  - export PATH=$PATH:$HOME/.local/bin
+  # login to ECR
+  - eval $(aws ecr get-login --no-include-email --region us-west-2)
 
 deploy:
-  provider: script
+  provider: heroku
   script:
     # push to heroku
     docker push registry.heroku.com/truthtree/web;
     heroku container:release web --app=truthtree;
   on:
-    branch: develop
\ No newline at end of file
+    branch: develop
+  provider: aws
+  script:
+    docker push $IMAGE_REPO_URL:latest;
+    ecs-deploy -c $CLUSTER_NAME -n $SERVICE_NAME -i $IMAGE_REPO_URL:latest -r us-west-2 --timeout 600;
+  on:
+    branch: master
```

Comment:
At which stage do we merge `develop` to `master` or does that happen automatically?

Reply:
It happens when we manually merge `develop` to `master`.